code/lib/cli/src/utils/legacy_storybookVersionDetection.ts

```diff
@@ -0,0 +1,59 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { sync as spawnSync } from 'cross-spawn';
+import semver from 'semver';
+
+const legacy_versionRegex = /(@storybook\/[^@]+)@(\S+)/;
+
+const legacy_excludeList = [
+  '@storybook/linter-config',
+  '@storybook/design-system',
+  '@storybook/ember-cli-storybook',
+  '@storybook/semver',
+  '@storybook/eslint-config-storybook',
+  '@storybook/bench',
+  '@storybook/addon-bench',
+  '@storybook/addon-console',
+  '@storybook/csf',
+  '@storybook/storybook-deployer',
+  '@storybook/addon-postcss',
+  '@storybook/react-docgen-typescript-plugin',
+  '@storybook/babel-plugin-require-context-hook',
+  '@storybook/builder-vite',
+  '@storybook/mdx1-csf',
+  '@storybook/mdx2-csf',
+  '@storybook/expect',
+  '@storybook/jest',
+  '@storybook/test-runner',
+  '@storybook/testing-library',
+];
+const legacy_isCorePackage = (pkg: string) =>
+  pkg.startsWith('@storybook/') &&
+  !pkg.startsWith('@storybook/preset-') &&
+  !legacy_excludeList.includes(pkg);
+
+/**
+ * @deprecated This function has been deprecated. This is solely to support upgrading from SB6 (and possible SB7)
+ * I took this code from https://github.com/storybookjs/storybook/blob/v6.5.16/lib/cli/src/upgrade.ts#L29-L109
+ */
+export const legacy_getStorybookVersion = (): string | undefined => {
```

Comment:
Let's wait for the adjustments of `findInstallations` I did here: https://github.com/storybookjs/storybook/pull/26067/files#diff-afe49a7cb8e5ea8e3d02751da8f553064c9a6ab67a0a819ac8048a98bf43f506R133 and then reuse `packageManager.findInstallations` here.

Then the support for multiple package managers is guaranteed.

Reply:
I personally don't feel like we should spend a lot of time refining this as it's purely to temporarily support an upgrade from SB6 to SBX